analyzers/src/SonarAnalyzer.CSharp/Rules/MethodParameterUnused.cs

```diff
@@ -126,8 +126,8 @@ private void ReportUnusedParametersOnMethod(MethodContext declaration)
 
         private void ReportOnDeadParametersAtEntry(MethodContext declaration, IImmutableList<IParameterSymbol> noReportOnParameters)
         {
-            var bodyNode = (CSharpSyntaxNode)declaration.Body ?? declaration.ExpressionBody;
-            if (bodyNode == null || declaration.Context.Node.IsKind(SyntaxKind.ConstructorDeclaration))
+            var declarationNode = (CSharpSyntaxNode)declaration.MethodDeclaration ?? declaration.LocalFunctionDeclaration;
```

Comment:
Let's get rid of the need to cast to `CSharpSyntaxNode` in a followup PR

Reply:
Added a task to my calendar for this.
Not sure when I'll tackle it really :/

Reply:
That was done in this PR in the end. It was easier than expected.

Reply:
#8374 is what I meant. I originally thought it's much bigger, but the core effort was already done in the past.

Reply:
Right.
I thought the comment was dedicated to `MethodParameterUnused`

Reply:
Yep, I wasn't clear enough with my original comment.